server/service/service_appconfig.go

```diff
@@ -80,6 +81,22 @@ func (svc *Service) sendTestEmail(ctx context.Context, config *fleet.AppConfig)
 	return nil
 }
 
+func (svc *Service) makeTestJiraRequest(ctx context.Context, jiraSettings *fleet.JiraIntegration) error {
+	client, err := externalsvc.NewJiraClient(&externalsvc.JiraOptions{
+		BaseURL:           jiraSettings.URL,
+		BasicAuthUsername: jiraSettings.Username,
+		BasicAuthPassword: jiraSettings.Password,
+		ProjectKey:        jiraSettings.ProjectKey,
+	})
+	if err != nil {
+		return &badRequestError{message: fmt.Sprintf("jira integration request failed: %s", err.Error())}
```

Comment:
`sendTestEmail` currently returns a `mailError` (which my guess is the UI already has code to parse)
```go
// mailError is set when an error performing mail operations
type mailError struct {
	message string
}

func (e mailError) Error() string {
	return fmt.Sprintf("a mail error occurred: %s", e.message)
}

func (e mailError) MailError() []map[string]string {
	return []map[string]string{
		{
			"name":   "base",
			"reason": e.message,
		},
	}
}
```
Maybe we can do something of similar format for the JIRA test request error?

Reply:
It encodes pretty much the same way, but `mailError` generates a status 500 which is wrong and we should not replicate (should be fixed as part of the ticket about consistency in status responses is tackled):

```
	case badRequestErrorInterface:
		je := jsonError{
			Message: "Bad request",
			Errors:  baseError(e.Error()),
		}
		w.WriteHeader(http.StatusBadRequest)
		enc.Encode(je)

	case mailError:
		me := jsonError{
			Message: "Mail Error",
			Errors:  e.MailError(),
		}
		w.WriteHeader(http.StatusInternalServerError)
		enc.Encode(me)
```
(`e.Error()` for bad request returns the error message, and wrapped in a `baseError(...)` call it encodes the same as the result of `e.MailError()`).

Reply:
Gotcha. Agree!